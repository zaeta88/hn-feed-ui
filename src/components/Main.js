import React, { Component } from 'react';
import Stories from './Stories';
import { get, del } from '../js/fetcher';
import './Main.css';

class Main extends Component {
  constructor(props) {
    super(props);
    this.state = {
      stories: []
    };
  }

  componentWillMount() {
    this.loadStories();
  }

  loadStories = async () => {
    const result = await get('/stories');
    this.setState({
      stories: result
    });
  }

  handleDeleteRequest = async (story) => {
    let { stories } = this.state;
    await del(`/stories/${story._id}`);
    this.setState({
      stories: stories.filter((s) => { return s !==story})
    });
  }

  handleDelete = (story) => {
    this.handleDeleteRequest(story);
  }

  render() {
    return (
      <div className="wrapper-div" id="wrapper">
        <main id="main">
          <div className="container">
            <Stories 
              stories={this.state.stories}
              handleDelete={this.handleDelete}
            />
          </div>
        </main>
      </div>
    );
  }
}

export default Main;