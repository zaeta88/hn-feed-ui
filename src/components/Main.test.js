import React from 'react';
import { shallow } from 'enzyme';
import Main from './Main';


describe('React.Component Main', () => {
  it('should render correctly', () => {
    const component = shallow(<Main />);
    expect(component).toMatchSnapshot();
  });
});
