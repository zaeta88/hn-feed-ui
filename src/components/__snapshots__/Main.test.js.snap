// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React.Component Main should render correctly 1`] = `ShallowWrapper {}`;
